backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrFileSerializer.kt

```diff
@@ -4,18 +4,22 @@ import org.jetbrains.kotlin.backend.common.LoggingContext
 import org.jetbrains.kotlin.backend.common.serialization.DeclarationTable
 import org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer
 import org.jetbrains.kotlin.backend.konan.RuntimeNames
+import org.jetbrains.kotlin.backend.konan.llvm.neverRetainFor
 import org.jetbrains.kotlin.ir.declarations.IrClass
 import org.jetbrains.kotlin.ir.declarations.IrFunction
 import org.jetbrains.kotlin.ir.util.hasAnnotation
+import org.jetbrains.kotlin.konan.target.KonanTarget
 
 class KonanIrFileSerializer(
+        val target: KonanTarget,
         logger: LoggingContext,
         declarationTable: DeclarationTable,
         bodiesOnlyForInlines: Boolean = false
 ): IrFileSerializer(logger, declarationTable, bodiesOnlyForInlines) {
 
     override fun backendSpecificExplicitRoot(declaration: IrFunction) =
-            declaration.annotations.hasAnnotation(RuntimeNames.exportForCppRuntime)
+            if (declaration.neverRetainFor(target)) false else
```

Comment:
That makes IR serialization different for differing targets. 
Currently `stdlib` shares a single IR representation between targets.
That would make it incorrect for some of the targets.
We need a bit more flexible solution. Let's discuss tomorrow.

Reply:
The solution should better be platform (js, jvm, native) aware. 
Because between platforms there can be differences similar to those between targets.